content_scripts/hints.js

```diff
@@ -232,9 +301,23 @@ var Hints = (function(mode) {
             var link = $('<div/>').css('top', Math.max(pos.top - bof.top, 0)).css('left', left)
                 .css('z-index', z + 9999)
                 .data('z-index', z + 9999)
-                .data('label', hintLabels[i])
                 .data('link', this)
-                .html(hintLabels[i]);
+            // Grouping
+            if (runtime.conf.hintGroups === true) {
+                var groupNum = Math.floor(i / groups.groupLen);
+                var groupIndex = i % groups.groupLen;
+                var char = groups.groups[groupNum][groupIndex];
```

Comment:
Looks like here are some duplicate logic with `genGroups`. We could refine this.